_pytest/monkeypatch.py

```diff
@@ -106,6 +108,14 @@ def __init__(self):
         self._cwd = None
         self._savesyspath = None
 
+    @contextmanager
+    def context(self):
```

Comment:
Please add a docstring here, something along the lines:

```python
"""
Context manager that returns a new :class:`MonkeyPatch` object which
undoes any patching done inside the ``with`` block upon exit:

.. code-block:: python

    import functools
    def test_partial(monkeypatch):
        with monkeypatch.context() as m:
            m.setattr(functools, "partial", 3)

Useful in situations where it is desired to undo some patches before the test ends,
such as mocking ``stdlib`` functions that might break pytest itself if mocked (for examples
of this see `#3290 <https://github.com/pytest-dev/pytest/issues/3290>`_.
"""
```

Reply:
Fixed